rotkehlchen/globaldb/handler.py

```diff
@@ -381,71 +381,69 @@ def retrieve_assets(userdb: 'DBHandler', filter_query: 'AssetsFilterQuery') -> t
         # filtered from the frontend where we set a limit in the number of results that goes
         # in the range from 10 to 100 and this guarantees that the size of the subquery is small.
         underlying_tokens_query = (
-            f'SELECT parent_token_entry, address, token_kind, weight FROM globaldb.underlying_tokens_list '  # noqa: E501
-            f'LEFT JOIN globaldb.evm_tokens ON globaldb.underlying_tokens_list.identifier=evm_tokens.identifier '  # noqa: E501
+            f'SELECT parent_token_entry, address, token_kind, weight FROM underlying_tokens_list '
+            f'LEFT JOIN evm_tokens ON underlying_tokens_list.identifier=evm_tokens.identifier '
             f'WHERE parent_token_entry IN (SELECT identifier FROM ({query}))'
         )
-
+        should_skip = filter_query.ignored_assets_handling.get_should_skip_handler()
         with userdb.conn.read_ctx() as cursor:
-            globaldb = GlobalDBHandler()
-            globaldb._wal_checkpoint()
-            cursor.execute(
-                f'ATTACH DATABASE "{globaldb.filepath()!s}" AS globaldb KEY "";',
-            )
-            try:
-                # get all underlying tokens
-                for entry in cursor.execute(underlying_tokens_query, bindings):
-                    underlying_tokens[entry[0]].append(UnderlyingToken.deserialize_from_db((entry[1], entry[2], entry[3])).serialize())  # noqa: E501
-
-                cursor.execute(query, bindings)
-                for entry in cursor:
-                    asset_type = AssetType.deserialize_from_db(entry[1])
-                    data = {
-                        'identifier': entry[0],
-                        'asset_type': str(asset_type),
-                        'name': entry[4],
-                    }
-                    # for evm tokens and crypto assets
-                    common_data = {
+            ignored_assets = userdb.get_ignored_asset_ids(cursor)
+
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            # get all underlying tokens
+            for entry in cursor.execute(underlying_tokens_query, bindings):
+                underlying_tokens[entry[0]].append(UnderlyingToken.deserialize_from_db((entry[1], entry[2], entry[3])).serialize())  # noqa: E501
+
+            cursor.execute(query, bindings)
+            for entry in cursor:
+                if should_skip(entry[0], ignored_assets):
```

Comment:
maybe same comment here? Or hmmm perhaps find a more nice central single point to write this comment? I will 100% definitely come in the future and ask why are we doing it this way if there is no comment. Also a TODO to maybe improve it (can't think of how right now, late at night)